includes/validation/class-amp-validation-manager.php

```diff
@@ -1848,7 +1848,7 @@ public static function validate_url( $url ) {
 		$response = ltrim( $response );
 
 		// Strip HTML comments that may have been injected at the end of the response (e.g. by a caching plugin).
-		$response = preg_replace( '/<!--.*?-->\s*$/s', '', $response );
+		$response = preg_replace( '/}\s*?<!--.*?-->\s*$/', '}', $response );
```

Comment:
The removal of the `s` will cause a problem in that it will not match a comment like this:

```html
…
</body></html>
<!--
	generated 2 seconds ago
	generated in 1.134 seconds
	served from batcache in 0.003 seconds
	expires in 298 seconds
-->
```

I've added a failing test case to demonstrate this: 9f7ef9c03.

Reply:
Added the `s` flag in b7c7e293a3527ef91c7acb40ccc38b5d67626e16.